src/main/java/halberstam/weather/CurrentWeatherView.java

```diff
@@ -0,0 +1,56 @@
+package halberstam.weather;
+
+import halberstam.weather.fivedayforcast.FiveDayForcast;
+import halberstam.weather.fivedayforcast.List;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class CurrentWeatherView extends JComponent {
+
+    FiveDayForcast fiveDayForcast;
+
+    public CurrentWeatherView() {
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://api.openweathermap.org/")
+                .addConverterFactory(GsonConverterFactory.create())
+                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+                .build();
+
+        OpenWeatherMapService service = retrofit.create(OpenWeatherMapService.class);
+
+        FiveDayForcast fiveDayForcast = service.getFiveDayForcast("Passaic").blockingFirst();
+        this.fiveDayForcast = fiveDayForcast;
+    }
+
+    public void setForcast(FiveDayForcast fiveDayForcast) {
+        this.fiveDayForcast = fiveDayForcast;
+        repaint();
+    }
+
+    protected void paintComponent(Graphics g)
+    {
+        super.paintComponent(g);
+        int height = getHeight();
+        g.translate(0, height);
+
+
+
+        java.util.List<List> forcastList = fiveDayForcast.list;
```

Comment:
If you remove the code from the constructor, `fiveDayForecast` will be null.
So make sure you don't execute this if `fiveDayForecast` is null